app/models/stripe_record.rb

```diff
@@ -182,17 +182,20 @@ def self.amount_to_ruby(amount_stripe_denomination, iso_currency)
     amount_stripe_denomination.to_i
   end
 
-  def self.create_from_api(api_record, parameters, account_id, parent_record = nil)
-    StripeRecord.create!(
-      stripe_record_type: api_record.object,
-      parameters: parameters,
-      stripe_id: api_record.id,
-      amount_stripe_denomination: api_record.amount,
-      currency_code: api_record.currency,
-      stripe_status: api_record.status,
-      account_id: account_id,
-      parent_record: parent_record,
-    )
+  def self.create_or_update_from_api(api_record, parameters = nil, account_id = nil, parent_record = nil)
+    StripeRecord.find_or_initialize_by(stripe_id: api_record.id) do |new_record|
+      new_record.parameters = parameters
+      new_record.account_id = account_id
+      new_record.parent_record = parent_record
+      new_record.stripe_record_type = api_record.object
```

Comment:
Hypothetically speaking, could we move `account_id` and `stripe_record_type` directly into the `find` call's parentheses parameters?